### PR TITLE
replace rsc.io/letsencrypt in favour of golang.org/x/crypto

### PR DESCRIPTION
**Pull request by tariq1890**
As per the readme over [here](https://github.com/rsc/letsencrypt/blob/master/README), it's recommended to use acme/autocerts.

closes https://github.com/docker/distribution/pull/2915

### CODE REVIEW

**Comment by tariq1890** [comment]
requesting your review @caervs

**Comment by caervs** [comment]
@dmcgowan @thaJeztah any opinions on this vs #2915? From where I stand reducing our dependencies is always a win and as @tariq1890 points out this is recommended in the letsencrypt readme.

**Comment by tariq1890** [comment]
Also want to state that this PR will lay down the path for go mod migration :). I'll have that up once this is merged.

**Comment by mrueg** [comment]
See also: https://github.com/docker/distribution/pull/2530

**Comment by thaJeztah** [comment]
> any opinions on this vs #2915?

I'm definitely not attached to #2915 - it was just to get rid of the fork, but (as Derek commented on that PR); that code needed replacing, so if this one does the job, I'm all in favour of that.

**Comment by tariq1890** [comment]
It looks like this PR is a duplicate of #2530 which was filed more than a year ago. It would be really great to see this merged. I am happy to get the PR review process going for this in case we are unable to revive the thread in #2530.

**Comment by jdolitsky** [comment]
+1 this would be really nice to have. Currently dealing with some dependency hell related to needing to use the @dmcgowan fork of `rsc.io/letsencrypt`

**Comment by thaJeztah** [comment]
Are documentation changes needed for this change?

**Comment by tariq1890** [comment]
I don't believe so. This PR intended to preserve any previous behaviour.

**Comment by tariq1890** [comment]
@caervs Did the e2e tests pass for this one?

**Comment by tariq1890** [comment]
Can I please get some response on this?

**Comment by caervs** [comment]
@tariq1890 yes turns out this change needs a few edits to work properly. Please:

* Add `Prompt:     autocert.AcceptTOS,` to the manager as it requires a prompt function
* Set `tlsConf.NextProtos = append(tlsConf.NextProtos, acme.ALPNProto)` (required for verification with letsencrypt
* Add a note to the config docs that the `ca-certificates` package is required when using letsencrypt to trust the letsencrypt cert

In the meantime we really need a second review to get this in. @manishtomar ptal. Thanks.

**Comment by tariq1890** [comment]
@caervs Thanks for your review comments. This is ready for another round of review.

**Comment by caervs** [comment]
> @caervs Do we have tests to verify this?

@manishtomar I have manually tested with no cache in place and restarting with existing cache. I have not tested with existing cache from letsencrypt fork carrying over into new cache, but I think that'd be overkill. Worst case the workaround is either to remove the cache or change the cache directory.

File: vendor.conf [hunk 1]
```diff
@@ -31,21 +30,17 @@ github.com/prometheus/procfs cb4147076ac75738c9a7d279075a253c0cc5acbd
 github.com/Shopify/logrus-bugsnag 577dee27f20dd8f1a529f82210094af593be12bd
 github.com/spf13/cobra 312092086bed4968099259622145a0c9ae280064
 github.com/spf13/pflag 5644820622454e71517561946e3d94b9f9db6842
-github.com/xenolf/lego a9d8cec0e6563575e5868a005359ac97911b5985
 github.com/yvasiyarov/go-metrics 57bccd1ccd43f94bb17fdd8bf3007059b802f85e
 github.com/yvasiyarov/gorelic a9bba5b9ab508a086f9a12b8c51fab68478e2128
 github.com/yvasiyarov/newrelic_platform_go b21fdbd4370f3717f3bbd2bf41c223bc273068e6
-golang.org/x/crypto c10c31b5e94b6f7a0283272dc2bb27163dcea24b
+golang.org/x/crypto e84da0312774c21d64ee2317962ef669b27ffb41
```
Comment (manishtomar) [inline vendor.conf]: Why this version? The [commit](https://github.com/golang/crypto/commit/e84da0312774c21d64ee2317962ef669b27ffb41) was on Oct 2018. Should it be more recent one? Also there are no versions in that repo. Not sure what is best version to pick.

Comment (tariq1890) [reply vendor.conf]: I did it for parity with the Go version (1.11.x) used by travis to build the distribution project.

Comment (manishtomar) [reply vendor.conf]: Sounds good. We should update travis to 1.12 but thats a different PR.

**Pull request merged**